WordPress/src/main/java/org/wordpress/android/ui/reader/repository/ReaderRepositoryEvents.kt

```diff
@@ -32,3 +34,12 @@ sealed class ReaderRepositoryCommunication {
         return "${this.javaClass.simpleName})"
     }
 }
+
+sealed class ReaderDiscoverCommunication {
```

Comment:
@malinajirka - this is an interesting thought. imho, I like the idea of discover tasks because it's clear, SRP and allows us to react start to finish for a task that we care about. `Failure` has been removed, so that's no longer an option for the very reason you state. Let's leave it for this PR and then chat later as you suggested.